### DISCUSSION TITLE
Custom Digital Files upload after purchase by api

### DISCUSSION THREAD

**Discussion by omercanvural** [etsy/open-api#1301] [Ideas]
I would like to have the option to upload "Custom Digital Files" by API after they are purchased and custimized for the buyer to finilize the purchase. I could not find anything related to this.

**Comment by Wilderbill** [comment]
I am a long-time programmer, but new to Etsy. (Still waiting for my API key approval).
Could you just have some sort of a cron job which checks once an hour for new orders and fulfills.

**Reply by omercanvural** [reply]
I will get to it if this gets solved. I have no needs to check for orders if i can't fill it auto.

**Reply by westerlund** [reply]
> I will get to it if this gets solved. I have no needs to check for orders if i can't fill it auto.

Doesn't seem to be possible to fulfill digital downloads.

**Comment by AlexDelPab** [comment]
I need this too. Any updates on this scenario?

**Comment by Wilderbill** [comment]
I assume you have a Python-like program which is helping you maintain your shop. Can't you just have that script copy the final product to your disk space (like Google Drive) and then send the customer an email with the file address.

**Reply by AlexDelPab** [reply]
It is an automated background job written in .NET/C#. While I know how to securely save blobs and send them to the customer, that’s not the issue. I can only complete orders for digital products on Etsy by uploading the file in the chat. How can I achieve this?

**Comment by Wilderbill** [comment]
I asked AI and it thinks you are currently out of luck: "As of now, Etsy does not provide a direct API or official mechanism to upload files to Etsy Chat programmatically. All file uploads to Etsy Chat must be done manually through their platform to maintain the integrity and security of transactions. To stay compliant with Etsy policies, focus on streamlining your workflow for manual uploads rather than attempting full automation."

**Comment by ramibch** [comment]
You can try to make request to `getShopReceipts` endpoint, get the `buyer_email` and send him/her an Email.

https://developers.etsy.com/documentation/reference#operation/getShopReceipts

I think Etsy does not give you the  `buyer_email` by default, you have to ask for it and give a good reason about why you need it.

**Reply by ramibch** [reply]
https://github.com/etsy/open-api/discussions/1185

**Reply by ramibch** [reply]
https://github.com/etsy/open-api/discussions/1281
